src/components/Shared/Filters/FilterPanel.tsx

```diff
@@ -412,45 +416,21 @@ export const FilterPanel: React.FC<FilterPanelProps & BoxProps> = ({
                 return {
                   ...acc,
                   [key]: value.split(',').map((enumValue) => {
-                    switch (enumValue) {
-                      // Status
-                      case 'active':
-                        return ContactFilterStatusEnum.Active;
-                      case 'hidden':
-                        return ContactFilterStatusEnum.Hidden;
-                      case 'null':
-                        return ContactFilterStatusEnum.Null;
-                      case 'Appointment Scheduled':
-                        return ContactFilterStatusEnum.AppointmentScheduled;
-                      case 'Ask in Future':
-                        return ContactFilterStatusEnum.AskInFuture;
-                      case 'Call for Decision':
-                        return ContactFilterStatusEnum.CallForDecision;
-                      case 'Contact for Appointment':
-                        return ContactFilterStatusEnum.ContactForAppointment;
-                      case 'Cultivate Relationship':
-                        return ContactFilterStatusEnum.CultivateRelationship;
-                      case 'Expired Referral':
-                        return ContactFilterStatusEnum.ExpiredReferral;
-                      case 'Never Ask':
-                        return ContactFilterStatusEnum.NeverAsk;
-                      case 'Never Contacted':
-                        return ContactFilterStatusEnum.NeverContacted;
-                      case 'Not Interested':
-                        return ContactFilterStatusEnum.NotInterested;
-                      case 'Partner - Financial':
-                        return ContactFilterStatusEnum.PartnerFinancial;
-                      case 'Partner - Pray':
-                        return ContactFilterStatusEnum.PartnerPray;
-                      case 'Partner - Special':
-                        return ContactFilterStatusEnum.PartnerSpecial;
-                      case 'Research Abandoned':
-                        return ContactFilterStatusEnum.ResearchAbandoned;
-                      case 'Unresponsive':
-                        return ContactFilterStatusEnum.Unresponsive;
-                      default:
-                        return ContactFilterStatusEnum.Null;
-                    }
+                    // Status
+                    // Check if saved filter (enumValue) is either the status  (partner_financial) or (Partner - Financial)
+                    const s = Object.entries(statusMapForFilters)?.find(
```

Comment:
Or maybe `matchedStatus`?

```suggestion
                    const status = Object.entries(statusMapForFilters)?.find(
```